### PR TITLE
Starting missing shard tasks

### PR DESCRIPTION
**Pull request by tbhanu-amzn**
### Description
Start missing failed tasks
 
### Issues Resolved
start missing failed tasks
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by krishna-ggk — COMMENTED** [review]
Changes looks good. 

Can we also add tests?

**Comment by saikaranam-amazon** [comment]
Integration tests are tracked here: https://github.com/opensearch-project/cross-cluster-replication/pull/81
Once the ES Changes and IT are pushed to ES_7.10 branch. We will port to main branch

**Review by ankitkala — DISMISSED** [review]
LGTM

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt [hunk 1]
```diff
@@ -790,13 +806,30 @@ class IndexReplicationTask(id: Long, type: String, action: String, description:
             ShardReplicationExecutor.TASK_NAME, replicationParams)
     }
 
+    override fun onIndexShardClosed(shardId: ShardId, indexShard: IndexShard?, indexSettings: Settings) {
+    }
+
```
Comment (krishna-ggk) [inline src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt:812]: Can we add a comment as to why this override exists?

**Review by krishna-ggk — DISMISSED** [review]

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt (hunk 1 above)
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt]: Done

File: src/main/kotlin/org/opensearch/replication/task/CrossClusterReplicationTask.kt [hunk 2]
```diff
@@ -142,8 +147,15 @@ abstract class CrossClusterReplicationTask(id: Long, type: String, action: Strin
...
-    fun onIndexOrShardClosed(indexOrShardId: Any) {
-        cancelTask("$indexOrShardId was closed.")
+    open fun onIndexShardClosed(shardId: ShardId, indexShard: IndexShard?, indexSettings: Settings) {
+        // keeping the default behavior to cancel the task
+        cancelTask("$shardId was closed.")
+    }
+
+    open fun onIndexRemoved(indexService: IndexService,
+                            reason: IndicesClusterStateService.AllocatedIndices.IndexRemovalReason) {
+        // keeping the default behavior to cancel the task
```
[the remark this answers is not in the document]
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/task/CrossClusterReplicationTask.kt]: Done

File: src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt [hunk 3]
```diff
@@ -128,7 +135,10 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
                     }
                 }
                 replicationMetadataManager.deleteIndexReplicationMetadata(request.indexName)
-                AcknowledgedResponse(true)
+                listener.onResponse(AcknowledgedResponse(true))
+            } catch (e: Exception) {
+                log.error("Stop replication failed for index[${request.indexName}] with error $e")
+                listener.onFailure(e)
```
[the remark this answers is not in the document]
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt]: Done

File: src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt [hunk 4]
```diff
@@ -87,17 +87,23 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
...
+            try {
                 log.info("Stopping index replication on index:" + request.indexName)
                 val isPaused = validateStopReplicationRequest(request)
 
                 val updateIndexBlockRequest = UpdateIndexBlockRequest(request.indexName,IndexBlockUpdateType.REMOVE_BLOCK)
-                client.suspendExecute(UpdateIndexBlockAction.INSTANCE, updateIndexBlockRequest, injectSecurityContext = true)
+                val updateIndexBlockResponse = client.suspendExecute(UpdateIndexBlockAction.INSTANCE, updateIndexBlockRequest, injectSecurityContext = true)
+                if(!updateIndexBlockResponse.isAcknowledged) {
+                    throw OpenSearchException("Failed to remove index block on ${request.indexName}")
+                }
```
[the remark this answers is not in the document]
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt:99]: The transport action has the errors logs. Response doesn't have any fields for capturing the reason for failure but any failure should be caught in the outer catch block.
Leaving as is for now

**Review by krishna-ggk — APPROVED** [review]

**Review by naveenpajjuri — APPROVED** [review]
LGTM

**Pull request merged**